File: backend/effects/registry.py
import json
import importlib.resources

def _load_req(effect_id):
    try:
        # Modern way to load resources from a package
        pkg = f"backend.effects.{effect_id}"
        resource = f"{effect_id}_requirements.json"
        with importlib.resources.files(pkg).joinpath(resource).open('r') as f:
            return json.load(f)
    except Exception:
        return {}

from .acrylic import acrylic
from .GoL import GoL
from .clone import clone
from .damping import damping
from .heisenbrush import heisenbrush
from .heisenbrush2 import heisenbrush2
from .qdrop import qdrop

EFFECT_REGISTRY = {
    "acrylic": {"module": acrylic, "config": _load_req("acrylic")},
    "GoL": {"module": GoL, "config": _load_req("GoL")},
    "clone": {"module": clone, "config": _load_req("clone")},
    "damping": {"module": damping, "config": _load_req("damping")},
    "heisenbrush": {"module": heisenbrush, "config": _load_req("heisenbrush")},
    "heisenbrush2": {"module": heisenbrush2, "config": _load_req("heisenbrush2")},
    "qdrop": {"module": qdrop, "config": _load_req("qdrop")},
}

File: frontend/src/effects/registry.js
import acrylicConfig from './acrylic/acrylic_requirements.json';
import heisenbrushConfig from './heisenbrush/heisenbrush_requirements.json';
import heisenbrush2Config from './heisenbrush2/heisenbrush2_requirements.json';
import qdropConfig from './qdrop/qdrop_requirements.json';
import dampingConfig from './damping/damping_requirements.json';
import cloneConfig from './clone/clone_requirements.json';
import GoLConfig from './GoL/GoL_requirements.json';

export const EFFECT_REGISTRY = {
    acrylic: { config: acrylicConfig, importModule: () => import('./acrylic/acrylic.js') },
    heisenbrush: { config: heisenbrushConfig, importModule: () => import('./heisenbrush/heisenbrush.js') },
    heisenbrush2: { config: heisenbrush2Config, importModule: () => import('./heisenbrush2/heisenbrush2.js') },
    qdrop: { config: qdropConfig, importModule: () => import('./qdrop/qdrop.js') },
    damping: { config: dampingConfig, importModule: () => import('./damping/damping.js') },
    clone: { config: cloneConfig, importModule: () => import('./clone/clone.js') },
    GoL: { config: GoLConfig, importModule: () => import('./GoL/GoL.js') }
};
export default EFFECT_REGISTRY;
